Document HA for Access Request plugins (#24945)
gravitational/teleport: docs/pages/access-controls/access-request-plugins/index.mdx

Teleport Just-in-Time Access Requests allow users to receive temporary elevated
privileges by seeking consent from one or more reviewers, depending on your
configuration.

> With Teleport's Access Request plugins, users can request, approve, and deny

Reviewer: This isn't the case for most of our plugins like Slack and Teams.  Only JIRA AFAIK can do approve and deny from a user within the outside system.

Author: I'll use a more general wording

Teleport Just-in-Time Access Requests allow users to receive temporary elevated
privileges by seeking consent from one or more reviewers, depending on your
configuration.

With Teleport's Access Request plugins, users can request, approve, and deny
access without leaving your organization's existing messaging and project
management solutions.

## Plugin guides

(!docs/pages/includes/access-request-integrations.mdx!)

## Architecture

> Access Request plugins are self-contained programs that connect to the Teleport
> Auth Service's gRPC API in order to listen for audit events that relate to new
> or updated Access Requests. After processing an Access Request event, Access
> Request plugins interact with a third-party API (e.g., the Slack or PagerDuty
> APIs). 

Reviewer 2: Suggestion to reduce extraneous words.

```suggestion
Access Request plugins are self-contained programs that connect to the Teleport
Auth Service's gRPC API to listen for audit events relating to new or updated
Access Requests. After processing an Access Request event, Access Request plugins
interact with a third-party API (e.g., the Slack or PagerDuty APIs).
```

Access Request plugins can run within private networks that are isolated from
the Teleport Auth Service. To access the Auth Service API, they connect to the
Proxy Service, which establishes a reverse tunnel for the plugin to access the
Auth Service.

> You can run multiple instances of an Access Request plugin for high
> availability. To do so, run each instance in a separate availability zone. There
> is no need for additional configuration or load balancing, as plugins avoid
> creating duplicate requests to their third-party APIs.

Reviewer 2: Tightening:

```suggestion
By running multiple instances of an Access Request plugin in separate availability
zones, you can achieve high availability. Plugins avoid creating duplicate requests
to third-party APIs, so no load balancing is required.
```

Author: I've suggested:

```markdown
You can run multiple instances of an Access Request plugin for high availability
by deploying each instance in a separate availability zone. There is no need for
additional configuration or load balancing, as plugins avoid creating duplicate
requests to their third-party APIs.
```

In the sentence, "By running multiple instances of an Access Request plugin in separate availability
zones, you can achieve high availability," I think putting "you can achieve high availability" at the end of the sentence makes it less clear, since we have to get to the end to know what the sentence is about.
